### PR TITLE
sql: add support for `initcap` function

### PR DESCRIPTION
**Pull request by benesch**
The `initcap` function capitalizes the first letter of each word in a string. See the documentation contained within for a more precise description.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Add support for the `initcap` function.

### CODE REVIEW

File: doc/user/content/releases/v0.97.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (mjibson) [inline doc/user/content/releases/v0.97.md:1]: Is this file supposed to be here? Related to `version-added`?

File: doc/user/data/sql_funcs.yml [hunk 2]
```diff
@@ -307,6 +307,13 @@
   - signature: 'constant_time_eq(a: text, b: text) -> bool'
     description: Returns `true` if the strings are identical, otherwise returns `false`. The implementation mitigates timing attacks by making a best-effort attempt to execute in constant time if the strings have the same length, regardless of their contents.
 
+  - signature: 'initcap(a: text) -> text'
+    description: >-
+      Returns `a` with the first character of every word in upper case and all
+      other characters in lower case. Words are separated by non-alphanumeric
+      characters.
+    version-added: v0.97
```
Comment (mjibson) [inline doc/user/data/sql_funcs.yml:315]: what's this?

**Review by mjibson — APPROVED** [review]

**Review by morsapaes — APPROVED** [review]
Thanks for the quick turnaround on this one! :fast: Documentation LGTM.

File: doc/user/data/sql_funcs.yml (hunk 2 above)
Comment (morsapaes) [reply doc/user/data/sql_funcs.yml:315]: We recently introduced special syntax to handle not-yet-released feature documentation. See #25156 for context.

File: doc/user/content/releases/v0.97.md (hunk 1 above)
Comment (benesch) [reply doc/user/content/releases/v0.97.md:1]: Yep, exactly!

File: doc/user/data/sql_funcs.yml (hunk 2 above)
Comment (benesch) [reply doc/user/data/sql_funcs.yml:315]: And there's docs on what this does here: https://github.com/MaterializeInc/materialize/blob/main/doc/user/README.md#adding-documentation-for-new-features

**Pull request merged**